docs/adr/0001-implement-nested-facets-tagging-for-specialist-documents-as-a-single-dropdown.md

```diff
@@ -0,0 +1,71 @@
+# 1. Implement nested facets tagging for specialist documents as a single dropdown
+
+Date: 2025-02-14
+
+## Status
+
+Proposed
+
+## Related PRs
+
+- [Implemented as part of this PR](https://github.com/alphagov/specialist-publisher/pull/2923)
+
+## Context
+
+Nested facets are required by a department as part of their new finder. Their existing finder contains: 
+
+- For filtering, a "Facet" dropdown and a "Sub facet" dropdown to filter for documents. Users are able to filter through a single facet value lookup at a time. 
+- The documents are tagged with multiple Facet and Sub facet values
+
+We need to be able to replicate this behaviour with minimal efforts to meet a deadline. 
+
+## Decision
+
+What we have implemented:
+
+- On the publisher side for tagging, a single dropdown that merges the facets' values (main facet and sub facets, essentially leaf nodes)
+- When we send the document to Publishing API, we write logic into the document model to figure out whether the leaf node is a parent facet or sub-facet value in the dropdown, and split the metadata we send to Publishing API into its own facet and a separate sub-facet in `specialist-publisher/app/models/document.rb` 
+  - Reasons for this being if we send them as separate facets, we can keep Publishing API and Search API functionality unchanged. 
+- When we load the edit page or new page again, the document is loaded from Publishing API, with the facet and sub-facet being separate facets, but the dropdown element expects a combined leaf node dropdown as mentioned in the first point, so we have to merge them back together again into the main facet field in `specialist-publisher/app/controllers/documents_controller.rb`.
+
+## Consequences
+
+Pros: 
+
+- Restrict user input to only valid options with a single dropdown. 
+- This supports both multiple select entries and single select entries. 
+- The workflow discussed as part of analysis document works.
+
+Cons:
+
+- Logic to merge the parent facet values, sub-facet values at form phase, split them at send phase, merge them again at load phase might be a little convoluted.
+
+## Alternatives proposed
+
+### Have tagging done as 2 separate facets with validation 
+
+Even on publisher side, surface all of a facet's parents facets as one dropdown, and all of the sub-facets as a separate dropdown. As outlined on [this branch](https://github.com/alphagov/specialist-publisher/compare/intellectual-property-office-finder_2?expand=1).
```

Comment:
Suggest linking to the commit instead, as the branch can easily be deleted in future, but the commit should live forever (citation needed!)

```suggestion
Even on publisher side, surface all of a facet's parents facets as one dropdown, and all of the sub-facets as a separate dropdown. As outlined in [this commit](https://github.com/alphagov/specialist-publisher/commit/c7346db771df5ccafd5fff8f7f687a8211436f6a).
```